src/test.js

```diff
@@ -36,6 +36,7 @@ export default function Test( settings ) {
 	this.module = config.currentModule;
 	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
+	this.timeout = 0;
```

Comment:
`config.testTimeout` is undefined by default. Having this one be `0` feels inconsistent. On the other hand, I don't like undefined as default for `config.testTimeout`. `false` is probably a better fit here. `0` could also work although my first-impression expectation based on other software would be that 0 either means infinite waiting or that no waiting is allowed (e.g. any async will be considered an error). Whereas in our case it means no waiting. Although I suppose infinite waiting and no waiting are very similar in the case of qunit - except that issue #1132 exposes a difference, albeit through a bug we should fix separately.

If we go with `0` we should document that for this and for `config.testTimeout` and which ever value we pick, we should file an issue to also change `config.testTimeout`'s default in the next major release.